app/controllers/users_controllers.rb

```diff
@@ -0,0 +1,35 @@
+class UsersControllers < ApplicationController
+  before_action :require_admin, only: [:index, :show, :new, :create, :destroy]
+  before_action :require_user, only: [:edit, :update]
```

Comment:
only: %i(edit update)

Reply:
require_user does not really look rails authorization way.
before_action :authenticate_user!
But install devise gem firstly